doCNA/Genome.py

```diff
@@ -247,53 +232,97 @@ def segment_genome (self, m0 = 0, fb_alpha = Consts.FB_ALPHA):
             for seg in self.chromosomes[chrom].segments:
                 self.all_segments.append (seg)
                 
+        self.segment_filter = [((s.end - s.start)/1e6 > Consts.SIZE_THR) &\
+                                (s.centromere_fraction < Consts.CENTROMERE_THR) &\
+                                (s.parameters['ai'] < Consts.DIPLOID_AI_THR) &\
+                                (np.abs(s.parameters['m']/self.genome_medians['m0']-1) < Consts.DIPLOID_dCN_THR/2)    for s in self.all_segments] 
         
-        self.score_model_distance ()
-        self.score_clonality (size_thr = Consts.SIZE_THR, model_thr = Consts.MODEL_THR,
-                              dalpha = Consts.DSCORE_ALPHA, kalpha = Consts.KSCORE_ALPHA,
-                              k_thr = Consts.K_THR)
+        self.segments = [self.all_segments[i] for i in np.where(self.segment_filter)[0]]
         
+        data_for_scoring = np.array([(s.parameters['ai'], 2*s.parameters['m']/self.genome_medians['m0']-2) for s in self.segments])
+        self.scorer = Scoring.Scoring (data_for_scoring, self.logger)
+        ps = []
+        for seg in self.all_segments:
+            self.scorer.score_dipl(seg)
+            ps.append (seg.parameters['p_HE'])
+        
+        ps = np.array(ps)
+        ##FDRing threshold
```

Comment:
this pattern appears fairly commonly in the code, where you declare an empty list, and then explicitly convert the list to an array. I'm not sure of the impact on memory here, but I know you could save some memory by doing:
```
ps = np.zeros(len(self.segments))
for i, seg in enumerate(self.all_segments):
    self.scorer.score_dipl(seg)
    ps[i] = seg.parameters['p_HE'])
```
I have no idea what data is actually being generated here, or is this would actually work, but when you declare a list and then explicitly convert the list to a numpy array, it has to copy the list to a newly initialized array in memory, and then (hopefully) the garbage collector would clean up the old list. I say hopefully because the garbage collector has to be smart enough to determine that this list is no longer being used by the program. 

I don't recommend changing everything now, but this may be reasonable if any of these lists are sufficiently large, and you are following this pattern.